python_modules/dagster/dagster/_core/execution/job_backfill.py

```diff
@@ -58,10 +62,46 @@ def execute_job_backfill_iteration(
 
     partition_set = _get_partition_set(workspace_process_context, backfill)
 
+    # refetch in case the backfill status has changed
```

Comment:
would it make sense to refactor this piece of code into a separate function that's used for asset backfills too?

This logic looks very similar: https://github.com/dagster-io/dagster/blob/0acea44533cad9bd7e253b40357823aa2db12790/python_modules/dagster/dagster/_core/execution/asset_backfill.py#L1169-L1201

Reply:
yeah i thought the same thing, and then tried to do it and got stuck and gave up. i think there was a really slight difference between what the asset daemon iteration would need as the return type and the job daemon would need as the return type. I'll give it another half hour tomorrow to see if i can crack it because it would be really nice to not have this duplicated

Reply:
ok i think i got it, would appreciate another review pass though, just to get a second set of eyes on it